crates/bevy_transform/src/propagation.rs

```diff
@@ -1,116 +1,45 @@
 use crate::components::{GlobalTransform, Transform};
-use bevy_ecs::prelude::{Changed, Entity, Query, With, Without};
-use bevy_hierarchy::{Children, Parent};
-
-/// Update [`GlobalTransform`] component of entities based on entity hierarchy and
-/// [`Transform`] component.
-pub fn transform_propagate_system(
-    mut root_query: Query<
-        (
-            Option<(&Children, Changed<Children>)>,
-            &Transform,
-            Changed<Transform>,
-            &mut GlobalTransform,
-            Entity,
-        ),
-        Without<Parent>,
-    >,
-    mut transform_query: Query<(
-        &Transform,
-        Changed<Transform>,
-        &mut GlobalTransform,
-        &Parent,
-    )>,
-    children_query: Query<(&Children, Changed<Children>), (With<Parent>, With<GlobalTransform>)>,
-) {
-    for (children, transform, transform_changed, mut global_transform, entity) in
-        root_query.iter_mut()
-    {
-        let mut changed = transform_changed;
-        if transform_changed {
-            *global_transform = GlobalTransform::from(*transform);
-        }
+use bevy_hierarchy::ComputedGlobal;
 
-        if let Some((children, changed_children)) = children {
-            // If our `Children` has changed, we need to recalculate everything below us
-            changed |= changed_children;
-            for child in children {
-                let _ = propagate_recursive(
-                    &global_transform,
-                    &mut transform_query,
-                    &children_query,
-                    *child,
-                    entity,
-                    changed,
-                );
-            }
-        }
+impl ComputedGlobal for GlobalTransform {
+    type Local = Transform;
+
+    type ToPropagate = GlobalTransform;
+
+    #[inline(always)]
+    fn from_local(local: &Self::Local) -> Self {
+        (*local).into()
```

Comment:
I tend to prefer being explicit about type conversion. It's more verbose but it's also much more readable when the type being converted into is explicit:
```suggestion
        GlobalTransform::from(*local)
```